fix(metrics): update mobc to latest version

alencardc opened the pull request:
Fixes prisma/prisma#18760 and prisma/prisma#18761.

Since https://github.com/importcjj/mobc/pull/74 is merged, we can update the mobc to 0.8.2 to get the fix.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prisma/prisma-engines?pullRequest=4193) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prisma/prisma-engines?pullRequest=4193) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prisma/prisma-engines?pullRequest=4193) it.</sub>

Jolg42 commented:
Thank you 🙌🏼 

Note that this [test](https://github.com/prisma/prisma-engines/blob/main/query-engine/black-box-tests/tests/metrics/smoke_tests.rs) is failing and need to be updated:
```
thread 'metrics::smoke_tests::smoke_tests::expected_metrics_rendered' panicked at 'assertion failed: metrics.contains(\"prisma_pool_connections_open counter\")', query-engine/black-box-tests/tests/metrics/smoke_tests.rs:60:13

```

janpio commented on query-engine/metrics/src/lib.rs:
```diff
@@ -79,11 +79,18 @@ const ACCEPT_LIST: &[&str] = &[
     PRISMA_CLIENT_QUERIES_TOTAL,
-    PRISMA_DATASOURCE_QUERIES_DURATION_HISTOGRAM_MS,
     PRISMA_DATASOURCE_QUERIES_TOTAL,
+    PRISMA_POOL_CONNECTIONS_OPENED_TOTAL,
+    PRISMA_POOL_CONNECTIONS_CLOSED_TOTAL,
+    PRISMA_POOL_CONNECTIONS_OPEN,
+    PRISMA_POOL_CONNECTIONS_BUSY,
+    PRISMA_POOL_CONNECTIONS_IDLE,
```
Is it correct to add our "aliases" here?
(Might also apply to the 2 `wait` metrics - not sure)

alencardc replied:
When `describe_metrics()` is called, it emits the initial value of the metrics using the alias. However, the [registry subscriber](https://github.com/prisma/prisma-engines/blob/b52b336a9238cdecd9b62b305cc926a11ad1b72f/query-engine/metrics/src/registry.rs#L304C44-L304C44) will ignore them, because they aren't on the list of accepted metrics. So, currently, those metrics are only registered when mobc emits them.

janpio replied:
And I assume that has negative impacts, like them being missing in some initial output or similar?
(I know only half the words you used in your comment - I _should_ ask an Engineer to take a look here 😝)

alencardc replied:
Sorry for not being as clear as I'd like 😬. You're right: some metrics may be missing from the first few API outputs. For example, `prisma_pool_connections_closed_total` will only be present in the output after a connection is closed because it's the moment when mobc emits a value for it (using the name `mobc_pool_connections_closed_total`).

janpio replied:
Thanks. That seems pretty inelegant in our code then. We should revisit this.

miguelff replied:
Really good catch in here, @alencardc Thank you! This part of the codebase needs good shaking!

janpio commented on query-engine/black-box-tests/tests/metrics/smoke_tests.rs:
```diff
@@ -55,15 +55,21 @@ mod smoke_tests {
-            assert!(metrics.contains("prisma_client_queries_wait gauge"));
+            assert!(metrics.contains("prisma_pool_connections_opened_total counter"));
+            assert!(metrics.contains("prisma_pool_connections_closed_total counter"));
+            //gauges
```
Nit
```suggestion
            // gauges
```

janpio commented on query-engine/black-box-tests/tests/metrics/smoke_tests.rs:
```diff
@@ -55,15 +55,21 @@ mod smoke_tests {
-            assert!(metrics.contains("prisma_datasource_queries_duration_histogram_ms_bucket"));
+            assert!(metrics.contains("prisma_client_queries_active gauge"));
+            assert!(metrics.contains("prisma_client_queries_wait gauge"));
+            //histograms
```
Nit:
```suggestion
            // histograms
```

alencardc commented:
Thank you for letting me know, @Jolg42 ! And thank you for updating the branch, @janpio 😅

Fixed the failing test and added the following metrics that weren't being asserted:
* prisma_pool_connections_closed_total
* prisma_client_queries_duration_histogram_ms
* prisma_client_queries_wait_histogram_ms

Also, I noted that the list of accepted metrics didn't contain those metrics emitted by mobc, so despite being registered [here](https://github.com/prisma/prisma-engines/blob/main/query-engine/metrics/src/lib.rs), they weren't added to the [registry](https://github.com/prisma/prisma-engines/blob/b52b336a9238cdecd9b62b305cc926a11ad1b72f/query-engine/metrics/src/registry.rs#L304C44-L304C44) during the query engine initialization. Added the missing metrics to accepted list.

Jolg42 reviewed (approved):
Thank you 🙌🏼
